sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidCorrelateUnnestRel.java

```diff
@@ -184,6 +195,99 @@ public DruidQuery toDruidQuery(boolean finalizeAggregations)
       unnestFilterOnDataSource = null;
     }
 
+    final DruidRel<?> newLeftDruidRel;
+    final DruidQuery updatedLeftQuery;
+
+    if (unnestDatasourceRel.getInputRexNode().getKind() == SqlKind.FIELD_ACCESS) {
```

Comment:
nit: this could use some more comments to indicate what is going on since its a pretty big wall of code